### ISSUE TITLE
Additional annotations

### ISSUE THREAD

**Issue by lawremi** [gaborcsardi/argufy#22]
Here are two more ideas for annotations that are somewhat orthogonal to type checking and assertions:
1. Whether an argument is required or not. Some arguments without defaults can still be missing.
2. Whether an argument is eagerly or lazily evaluated.

This information would be useful for documentation, code analysis and generation, etc.

Related to this is the desire for a standard syntax that requests eager evaluation in a function call. For example, 

```
transform(x, new_column = ?expr)
```

The code generated for a lazy argument declaration could resolve that syntax automatically and force the promise instead of quoting it.

**Comment by gaborcsardi** [comment]
Yes, I agree that we could support required and not required arguments as well, although I am often annoyed by functions that use `missing(...)` because it is hard to use them programatically. I prefer a default value of `NULL`.

I guess argument checking mostly screws up lazy evaluation. I am not sure what we could do about this, or whether we should do anything at all. I am not entirely sure what you mean by the last sentence.

**Comment by lawremi** [comment]
These ideas are meant to be useful beyond the context of argument assertions. For example, in documentation. Or, what the last sentence mentioned: an auto-generated "escape hatch" for lazy evaluation.

**Comment by gaborcsardi** [comment]
Skipping this for now, will come back to it later.
